### DISCUSSION TITLE
Invoking() should allow null subjects

### DISCUSSION THREAD

**Discussion by AVee** [fluentassertions/fluentassertions#1732] [Q&A] [answered]
### Description
As per https://github.com/fluentassertions/fluentassertions/commit/15c20b34a45503c60c97f4660f2260e4e338f985 the Invoking() methods are rejecting null subjects. While I fully agree with the check for the action, I feel that a null subject should be valid. 

E.g. this seems a sensible test to write:

```csharp
        public static void ThrowOnNull(object? arg)
        {
            if (arg == null)
                throw new ArgumentNullException(nameof(arg));
        }

        [Fact]
        public void ThrowsOnNull()
        {
            object? arg = null;
            arg.Invoking(x => ThrowOnNull(x))
                .Should()
                .Throw<ArgumentNullException>();
        }
```

### Expected behavior:

The above test should succeed.

### Actual behavior:

Currently this test fails, with an System.ArgumentNullException from within FluentAssertions.

### Versions

FluentAssertions 6.2.0

**Comment by dennisdoomen** [comment, marked as answer]
The intention of `Invoking` is to invoke members of the subject-under-test. So we don't expect the subject itself to ever be `null`. But you can rewrite this as:

```csharp
Action act = () => ThrowOnNull(arg);
act.Should().Throw<ArgumentNullException>();
```

**Reply by jnyrup** [reply]
Alternatively, this might fit your desire.

```cs
using static FluentAssertions.FluentActions;

Invoking(() => ThrowOnNull(null)).Should().Throw<ArgumentNullException>();
```
